bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageWebTargets.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.opengarage.internal;
+
+import java.io.IOException;
+
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.opengarage.internal.api.ControllerVariables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles performing the actual HTTP requests for communicating with Opengarage units.
+ *
+ * @author Paul Smedley - Initial Contribution
+ *
+ */
+public class OpenGarageWebTargets {
+    private static final int TIMEOUT_MS = 30000;
+
+    private String getControllerVariablesUri;
+    private String changeControllerVariablesUri;
+    private Logger logger = LoggerFactory.getLogger(OpenGarageWebTargets.class);
+
+    public OpenGarageWebTargets(String ipAddress, long port, String password) {
+        String baseUri = "http://" + ipAddress + ":" + port + "/";
+        getControllerVariablesUri = baseUri + "jc";
+        changeControllerVariablesUri = baseUri + "cc?dkey=" + password;
+    }
+
+    public ControllerVariables getControllerVariables() throws OpenGarageCommunicationException {
+        String response = invoke(getControllerVariablesUri);
+        return ControllerVariables.parse(response);
+    }
+
+    public void setControllerVariables(Boolean request) throws OpenGarageCommunicationException {
+        logger.debug("Received request: {}", request);
+          if (request.equals(true)) {
+              logger.debug("Received request to open door");
+              String queryParams = "&open=1";
+              invoke(changeControllerVariablesUri, queryParams);
+          } else {
+              logger.debug("Received request to close door");
+              String queryParams = "&close=1";
+              invoke(changeControllerVariablesUri, queryParams);
+          }
+    }
+
+    private String invoke(String uri) throws OpenGarageCommunicationException {
+        return invoke(uri, "");
+    }
+
+    private String invoke(String uri, String params) throws OpenGarageCommunicationException {
+        String uriWithParams = uri + params;
+        logger.debug("Calling url: {}", uriWithParams);
+        String response;
+        synchronized (this) {
+            try {
+                response = HttpUtil.executeUrl("GET", uriWithParams, TIMEOUT_MS);
+            } catch (IOException ex) {
```

Comment:
You might want to log here. Either to debug or trace. Otherwise If a user has a problem you have no way of finding out what the problem is.

Reply:
The URL is logged above, this is what I used when debugging it locally?

Reply:
I mean if some user using openHAB will trigger a problem here and that user reports on the forum it doesn't work. Because there is no logging it's not clear at first the problem originates here and when you want to help (and can't reproduce it locally) you have no way of finding out what is causing the problem because the error is suppressed.

Reply:
OK, see 7785e06b93